### ISSUE TITLE
Support for self-issued 

### ISSUE THREAD

**Issue by CrowdHailer** [DockYard/openid_connect#26]
https://openid.net/specs/openid-connect-core-1_0.html#SelfIssued

I can't find any documentation around how to do self issued identity providers. I don't think it works by just setting the correct issuer. Is there a setting or is it unsupported

**Comment by bcardarella** [comment]
I didn't require self issued for my needs so it hasn't yet been implemented.

**Comment by bcardarella** [comment]
It seems like it would be trivial to implement though

**Comment by CrowdHailer** [comment]
Yeah. I've just finished walking through the Process in JS land.
These were all the changes that were required. https://github.com/panva/node-openid-client/commit/c5d315826a767d1479509931eddb5ae6e3b99532

Might be a useful resource if only for the tests.

**Comment by CrowdHailer** [comment]
Would you be interested in adding support if I opened a PR?

**Comment by bcardarella** [comment]
Yes definitely, I started work this AM but got pulled onto other things. If you're offering I'm taking :)

**Comment by CrowdHailer** [comment]
I'm still trying to work my way around the library. If you had some
pointers on where I should start that would be great. Hopefully have more
time to work on it next week.
